File: practica 1/sesion 2/DS_P1S2-FactoriaAbstracta-MetodoFactoriaJava/src/ds_p1_sesion2/Sistema.java
package ds_p1_sesion2;

public class Sistema {
	public FactoriaUsuarios factoriaUsuarios;
	public Cliente cliente;
	public Tecnico tecnico;
        private static MenuSistema menu = new MenuSistema();
        
        private static Trabajo configurarTrabajoPremium(String nombre_cliente, String descripcion_trabajo, float precio, String nombre_tecnico, boolean enActivo){
           FactoriaUsuarios creadorusuario;
           creadorusuario = new FactoriaUsuariosPremium();
           Trabajo trabajo = new Trabajo(creadorusuario.crearCliente(nombre_cliente), descripcion_trabajo, precio);
           
           return trabajo;
        }
        
        private static Trabajo configurarTrabajoLite(String nombre_cliente, String descripcion_trabajo, float precio, String nombre_tecnico, boolean enActivo){
           FactoriaUsuarios creadorusuario;
           creadorusuario = new FactoriaUsuariosLite();
           Trabajo trabajo = new Trabajo(creadorusuario.crearCliente(nombre_cliente), descripcion_trabajo, precio);
           return trabajo;
        }
        
        public static void main(String[] args) {
            Trabajo premium = configurarTrabajoPremium("Ahmed","Arreglar lavabo", (float) 2000.0, "Juan", true);
            Trabajo lite = configurarTrabajoLite("Iván","Alicatar", (float) 5000.0, "Ignacio", false);
            System.out.println(premium.toString());
            System.out.println(lite.toString());
            
            menu.menuInicial();
        }
}

File: practica 1/sesion 2/DS_P1S2-FactoriaAbstracta-PrototipoRuby/Sistema.rb
require('./FactoriaPrincipal.rb');
require('./FactoriaPrototipoLite.rb');
require('./FactoriaPrototipoPremium.rb');
require('./Cliente.rb');
require('./Tecnico.rb');
require('./Trabajo.rb')

module DSPrototype
public
class Sistema

	public
	def self.configurarTrabajoPremium(nombre_cliente,nombre_empleo, precio, nombre_tecnico, enActivo)
    factoria = FactoriaPrototipoPremium.new()
	  trabajo = Trabajo.createTrabajoFactoria(factoria,nombre_cliente,nombre_empleo,precio)
	  trabajo.solicita.push(factoria.crearTecnico(enActivo, nombre_tecnico))
	  return trabajo
	end
	
  def self.configurarTrabajoLite(nombre_cliente,nombre_empleo, precio, nombre_tecnico, enActivo)
    factoria = FactoriaPrototipoLite.new()
    trabajo = Trabajo.createTrabajoFactoria(factoria,nombre_cliente,nombre_tecnico,precio)
    trabajo.solicita().push(factoria.crearTecnico(enActivo, nombre_tecnico))
    return trabajo
  end	
	
	def initialize()
		@_unnamed_FactoriaPrincipal_
		@_unnamed_Cliente_
		@_unnamed_Tecnico_
	end
	
	def self.main
    print("\nCreando Premium")
	  premium = self.configurarTrabajoPremium("Ahmed","Arreglar lavabo", 2000.0, "Juan", true)
    print("\nCreando Lite")
	  lite = self.configurarTrabajoLite("Iván","Alicatar", 5000.0, "Ignacio", false)
	  print("\n#{premium.toString()}")
    print("\n#{lite.toString()}")
	end
end
end

DSPrototype::Sistema.main
